CMakeLists.txt

```diff
@@ -57,8 +57,11 @@ if(NOT MSVC)
     add_compile_options(-fPIC -Wall -Wpedantic
         $<$<CXX_COMPILER_ID:GNU>:-fdiagnostics-color=always>
         $<$<CXX_COMPILER_ID:Clang,AppleClang>:-fcolor-diagnostics>)
+    if (CMAKE_BUILD_TYPE STREQUAL "Release")
+        add_compile_options(-D_FORTIFY_SOURCE=2)
```

Comment:
Please, verify if optimization level of 1 and above has to be set for this macro to work as expected.

Reply:
FORTIFY_SOURCE requires O>0, we use O3 for the Release version.